cram/cram_decode.c

```diff
@@ -2368,10 +2365,10 @@ int cram_decode_slice(cram_fd *fd, cram_container *c, cram_slice *s,
 	if ((!s->ref && s->hdr->ref_base_id < 0)
 	    || memcmp(digest, s->hdr->md5, 16) != 0) {
 	    char M[33];
-	    fprintf(stderr, "ERROR: md5sum reference mismatch for ref "
-		    "%d pos %d..%d\n", ref_id, s->ref_start, s->ref_end);
-	    fprintf(stderr, "CRAM: %s\n", md5_print(s->hdr->md5, M));
-	    fprintf(stderr, "Ref : %s\n", md5_print(digest, M));
+	    hts_log_error("MD5 checksum reference mismatch for ref %d pos %d..%d",
+		ref_id, s->ref_start, s->ref_end);
+	    hts_log_error("CRAM: %s", md5_print(s->hdr->md5, M));
```

Comment:
I had a look at htslib's sam.c and see it has a mix of ```%s```, ```"%s"``` and ```'%s'```, depending on message.

These error messages were written by a different author, initially in a different project, to the htslib ones so there is probably an inconsistency of style.   I have no strong preferences on the style though and feel free to improve wording if you spot anything phrased poorly too.  Thank you @anderskaplan for your work on logging.